client_authentication.go

```
@@ -251,6 +262,43 @@ func (f *Fosite) DefaultClientAuthenticationStrategy(ctx context.Context, r *htt
 	return client, nil
 }
 
+// Tries to read client ID from (yet) unverified token, if request came unauthenticated
+func jwtBearerNoClientAuthRequest(r *http.Request, form url.Values, originalErr error) (clientID, clientSecret string, err error) {
+	if grantType := form.Get("grant_type"); grantType != grantTypeJWTBearer {
+		return "", "", originalErr
+	}
+	assertion := form.Get("assertion")
+
+	if assertion == "" {
+		return "", "", errorsx.WithStack(ErrInvalidRequest.WithHint("Required parameter assertion is missing."))
```

Please return the original error here